packages/iris-grid/src/IrisGridTableModel.js

```diff
@@ -74,6 +74,8 @@ class IrisGridTableModel extends IrisGridModel {
     // These rows can be sparse, so using a map instead of an array.
     this.pendingNewDataMap = new Map();
     this.pendingNewRowCount = 0;
+
+    this.frozenColumns = [];
```

Comment:
Think of the `frozenColumns` array more of a `userFrozenColumns`, we only want to set it when the user explicitly sets it. So we should just initialize it to `this.frozenColumns = null`. Otherwise, we would save the layoutHints frozen columns into the workspace, and then when reloading we would always use that, even if the layout hints have changed.

One easy way to deal with this could be having a function like:

```
get frozenColumns() {
    return this.userFrozenColumns ?? this.layoutHints?.frozenColumns ?? [];
}
```
Or something like that